cpp/demo/checkpointing/main.cpp

```diff
@@ -40,6 +40,13 @@ int main(int argc, char* argv[])
     adios2::Engine engine = io.Open("mesh.bp", adios2::Mode::Append);
 
     io::native::write_mesh(io, engine, *mesh);
+    std::span<float> x = mesh->geometry().x();
+    for (std::size_t i = 0; i < x.size(); ++i)
+    {
+      x[i] *= 4;
+    }
```

Comment:
```suggestion
    for (std::size_t i = 0; i < x.size(); ++i)
    {
      x[i] *= 4;
    }
```
We could use `std::for_each` to illustrate good coding practices, see:
https://godbolt.org/z/97osavhb6